src/main/java/core/basesyntax/service/impl/DataParserImpl.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataParserImpl implements DataParser {
+    private static final String SEPARATOR = ",";
+    private static final int INFORMATION_LINE_INDEX = 0;
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int FRUIT_QUANTITY_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> parse(List<String> lines) {
+        lines.remove(INFORMATION_LINE_INDEX);
+        List<FruitTransaction> transactions = new ArrayList<>();
+        lines = removeSpaces(lines);
+        for (String line : lines) {
+            transactions.add(parseFruitTransaction(line));
+        }
+        return transactions;
+    }
+
+    private FruitTransaction parseFruitTransaction(String line) {
+        String[] elements = line.split(SEPARATOR);
+        FruitTransaction.Operation operation =
+                FruitTransaction.Operation.fromCode(elements[OPERATION_INDEX]);
+        String name = elements[FRUIT_NAME_INDEX];
+        int quantity = Integer.parseInt(elements[FRUIT_QUANTITY_INDEX]);
+        return new FruitTransaction(name, quantity, operation);
+    }
+
+    private List<String> removeSpaces(List<String> lines) {
+        lines.replaceAll(str -> str.replace(" ", ""));

```

Comment:
The 'removeSpaces' method is currently removing all spaces from the strings. This might not be safe if the data can contain valid spaces within the content. If you're only intending to trim whitespace from the beginning and end of the strings, consider using 'str.trim()' instead.